spring-rabbit/src/test/java/org/springframework/amqp/rabbit/listener/MessageListenerContainerLifecycleIntegrationTests.java

```diff
@@ -115,7 +119,7 @@ public int value() {
 		}
 	}
 
-	@Rule
+//	@Rule
```

Comment:
I think we need to treat a container shutdown different to a context destroy.

It seems to me that the issue (creating a new connection) is only a problem if the connection factory has been destroyed by a context `close()`. Otherwise, it doesn't hurt to try to send a late ack.

Maybe we need to make the CCF look at the context state and only re-create the connection if the context is active.